### PR TITLE
MC-31518: Price filters do not appear if the Show out of stock option…

### PR DESCRIPTION
**Pull request by tsviklinskyi**
### Description (*)
https://jira.corp.magento.com/browse/MC-31518
ISSUE
The category page does not display a price filter if the Show out of stock option is enabled and a non-default source is used. This is reproduced if the page contains only configurable products. If there are also simple products, the filter appears, but shows incorrect values. The reason is the values min_price=0 and max_price=0 in the catalog_product_index_price table.

STEPS TO REPRODUCE

Take 2.3-develop + MSI
Create a new source: Stores->Inventory->Sources, enter Name, Code, Address. Save.
Create a new Stock: Stores->Inventory->Stocks, enter Name, Website = Main, assign source from previous step
Create a category
Create two configurable products: Enter the Name, SKU, Price $100 for first conf product and $200 for second conf product, Category from #4, Create configuration, choose Color, add two options (black and white), Generate products, Save.
Open the added options, Assign source from step #2 (Do not add a default). Specify the quantity of 100 pieces. Save.
Do indexing if necessary
Go to frontend, open category from #4. You will see two configurable products and a price filter on the left
Go to backend. Stores->Configuration->Catalog->Inventory->Stock Options, set Display Out of Stock Products to Yes. Save
Do indexing
Go to frontend, open category from #4.
EXPECTED RESULTS:
Price filter should be visible in Layered Navigation.

ACTUAL RESULTS:
Price filter is missing.

Updated 2020-03-16:

STEPS TO REPRODUCE
1. Create a second website (code "website2"), do not mark it as "default". The Main Website should remain as the default website.
2. Go to Stores > Configuration > Catalog > Inventory > Stock Options, set Display Out of Stock Products to Yes
3. Go to Stores > Catalog > Catalog > Price > Catalog Price Scope to Website
4. Go to Stores > Inventory > Sources and create a second source: "source2"
5. Go to Stores->Inventory->Stocks and create a second Stock: "stock2"

Assign stock to second website "website2"
Assign stock to newly created source "source2"

6. Create a category
7. Create 2 different configurable products

config1, price $150, two color options black & blue
config2, price $49, two color options black & blue
Assign products to "website2"
8. Open configurable options for each configurable product

Make sure they are assigned to "website2"
Assign "source2" and add any stock qty > 0
9. Go to category from step #6 and assign both configurable products and their children to this category
10. reindex
11. Open category in Main Website
 No products are displayed as products are assigned to Second Website ("website2")
12. Open category in website2
 Price Filter is still not showing.

### Related Pull Requests
- https://github.com/magento-tsg-csl3/magento2ce/pull/224

### Fixed Issues (if relevant)

1. magento/inventory#<issue_number>: Issue title

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by tsviklinskyi** [comment]
Fix was changed.
Created PR: https://github.com/magento-tsg-csl3/magento2ce/pull/224

**Comment by tsviklinskyi** [comment]
run all tests
with extension

**Comment by tsviklinskyi** [comment]
run all tests
with extensions

**Comment by tsviklinskyi** [comment]
run all tests
with extensions

**Comment by serhii-balko** [comment]
run all tests
with extensions

File: InventoryConfigurableProduct/Plugin/Model/ResourceModel/Product/Indexer/Price/Configurable/FilterSelectByInventoryPlugin.php [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    public function __construct(
+        StoreManagerInterface $storeManager,
+        StockResolverInterface $stockResolver,
+        StockIndexTableNameResolverInterface $stockIndexTableNameResolver,
+        DefaultStockProviderInterface $defaultStockProvider = null
+    ) {
+        $this->storeManager = $storeManager;
+        $this->stockResolver = $stockResolver;
+        $this->stockIndexTableNameResolver = $stockIndexTableNameResolver;
+        $this->defaultStockProvider = $defaultStockProvider ?: ObjectManager::getInstance()
```
Comment (sidolov) [inline InventoryConfigurableProduct/Plugin/Model/ResourceModel/Product/Indexer/Price/Configurable/FilterSelectByInventoryPlugin.php]: You don't need to use ObjectManager in newly created code.

File: InventoryConfigurableProduct/Plugin/Model/ResourceModel/Product/Indexer/Price/Configurable/FilterSelectByInventoryPlugin.php [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        $websiteCode = $this->storeManager->getWebsite()->getCode();
+        $stock = $this->stockResolver->execute(SalesChannelInterface::TYPE_WEBSITE, $websiteCode);
+        $stockId = (int)$stock->getStockId();
+        if ($stockId === $this->defaultStockProvider->getId()) {
+            return $select;
+        }
+        $stockTable = $this->stockIndexTableNameResolver->execute($stockId);
+
+        $whereParts = $select->getPart(Select::WHERE);
+        foreach($whereParts as $key => $cond){
```
Comment (sidolov) [inline InventoryConfigurableProduct/Plugin/Model/ResourceModel/Product/Indexer/Price/Configurable/FilterSelectByInventoryPlugin.php]: It's weird to iterate where condition and remove statements from it. Is it possible to add new logic in another way?

**Review by sidolov — CHANGES_REQUESTED** [review]

**Comment by sidolov** [comment]
@tsviklinskyi please, cover changes with integration or functional tests. Thank you!

**Comment by sidolov** [comment]
Close due to significant unacceptable performance degradation for all merchants using MSI.

**Pull request closed without merging**